src/CSharp/CodeCracker/Style/ForInArrayCodeFixProvider.cs

```diff
@@ -50,10 +53,27 @@ where controlVarId.Equals(initSymbol)
                                      select s).ToList();
             var arrayDeclaration = arrayDeclarations.First();
             var blockForFor = forBlock.RemoveNode(arrayDeclaration, SyntaxRemoveOptions.KeepLeadingTrivia);
+
             var forEachStatement = SyntaxFactory.ForEachStatement(SyntaxFactory.ParseTypeName("var"), arrayDeclaration.Declaration.Variables.First().Identifier, arrayAccessor.Expression, blockForFor)
                 .WithLeadingTrivia(forStatement.GetLeadingTrivia())
                 .WithTrailingTrivia(forStatement.GetTrailingTrivia())
                 .WithAdditionalAnnotations(Formatter.Annotation);
+
+            var foreachVariable = SyntaxFactory.IdentifierName(forEachStatement.Identifier.Text);
+            Func<ForEachStatementSyntax, List<ElementAccessExpressionSyntax>> getElementsAccessorsToReplace;
+            getElementsAccessorsToReplace = (foreachStatement) => forEachStatement.DescendantNodes()
+                            .OfType<ElementAccessExpressionSyntax>()
+                            .Where(eae => eae.ArgumentList.Arguments.Any(a => a.Expression.ToFullString().Trim() == forVariable.Identifier.ToFullString().Trim()))
+                            .ToList();
```

Comment:
@AlbertoMonteiro I found that that last was not correct, as it would introduce a compilation error. I changed the analyzer to disallow accessors that are not from the same symbol. Other types of use of the for initializer variable were already forbidden. This bug was there already before you worked your fix.